server/etcdserver/api/rafthttp/hijackedReadCloser.go

```diff
@@ -0,0 +1,54 @@
+// Copyright 2024 The etcd Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rafthttp
+
+import (
+	"io"
+	"net/http"
+)
+
+type hijackedReadCloser struct {
+	originalReadCloser io.ReadCloser
+}
+
+func (h *hijackedReadCloser) Read(p []byte) (int, error) {
+	// When hijacking, we read the data but drop it completely
+	// gofail: var HijackRequestBodyFailPoint struct{}
```

Comment:
you don't need to inject gofail point here. Just drop the data. Because this is only called inside failpoint.

Reply:
We actually need this. I probably should have done a better job in the comment. 

The hijacking of the `RoundTrip` and `ServeHTTP` is to use a custom version of them respectively. The custom version basically contains a custom `Reader` and `Writer`, when the failpoint is not triggered, the read and write operation will go through normally, but when triggered all data traffic will be dropped. 

Thus, there are 2 types of failpoints
- the first type is `HijackRoundTripperFailPoint`, `HijackSnapshotHandlerFailPoint`, `HijackPipelineHandlerFailPoint`, and `HijackStreamHandlerFailPoint`, which are the failpoints that should be enabled when the etcdprocess is started (which is currently a bug as I can't manage to do so), since it will let the code go through custom version of `RoundTrip` and `ServeHTTP` to prepare for the hijacking action
- the second type is `HijackRequestBodyFailPoint` and `HijackResponseWriterFailPoint`, which should be enabled and disabled alongside with the `blockhole`

Does this make sense? :)